src/custom/hooks/useRefetchPriceCallback.ts

```diff
@@ -35,23 +46,42 @@ export function useRefetchQuoteCallback() {
         : undefined
 
       const [fee, price] = await Promise.all([feePromise, pricePromise])
-      if ((fee || !fetchFee) && price) {
+      // use typecheck on returned fee/price response
+      // can be a valid FeeInformation/Price object
+      // or an UnsupportedToken error response
+      if ((_validResponse(fee) || !fetchFee) && _validResponse(price)) {
```

Comment:
I think we should merge my new PR, now this is simpler and uses a try-catch. 

I think you should handle here in the catch the error of UnsupportedToken.
I would just check the error code.